### PR TITLE
splitting mutation types using metadata

### PR DESCRIPTION
**Pull request by mufernando**
With popsim-consortium/stdpopsim#1152, now we store information about the DFEs and mutation types in the metadata of the tree sequence.

In this PR, I changed the way we split sites by class (neutral vs. non-neutral) using the metadata.

working on this with @izabelcavassim 

PS: this is a re-write of #36

### CODE REVIEW

**Comment by mufernando** [comment]
@andrewkern are you running the pipeline on talapas? if so, do we have a slurm config file that works with talapas in here? I'm not sure how to test that whatever I changed didn't mess things up.

**Comment by andrewkern** [comment]
i've been running on sesame. i've tuned the defaults in the config so that everything runs very quickly for debugging (scaling, etc)

**Comment by mufernando** [comment]
I think this is ready to be merged? I don't have access to the tuned config file that you mentioned @andrewkern, so could you merge this and rebase your working branch and test that everything works well? I don't see how this wouldn't work.

File: benchmark/ts2fs.py [hunk 1]
```diff
@@ -19,33 +19,44 @@ def f(x):
...
             for md in m.metadata["mutation_list"]:
                 mt.append(md["mutation_type"])
-        if len(set(mt)) > 1:
-            mut_type[j] = 3
-        else:
-            mut_type[j] = mt[0]
+        site_class[j] = mut_types[mt[0]] if len(mt) == 1 else "double_hit" 
+    assert sum(site_class == None) == 0
+    print("Number of sites per class is:", flush=True)
+    unique, counts = numpy.unique(site_class, return_counts=True)
```
Comment (andrewkern) [inline benchmark/ts2fs.py:42]:
```suggestion
    unique, counts = np.unique(site_class, return_counts=True)
```

File: benchmark/ts2fs.py [hunk 2]
```diff
@@ -19,33 +19,44 @@ def f(x):
...
-            # print(m)
             for md in m.metadata["mutation_list"]:
                 mt.append(md["mutation_type"])
-        if len(set(mt)) > 1:
-            mut_type[j] = 3
-        else:
-            mut_type[j] = mt[0]
+        site_class[j] = mut_types[mt[0]] if len(mt) == 1 else "double_hit" 
+    assert sum(site_class == None) == 0
+    print("Number of sites per class is:", flush=True)
```
Comment (andrewkern) [inline benchmark/ts2fs.py:41]:
```suggestion
    # print("Number of sites per class is:", flush=True)
```

File: benchmark/ts2fs.py [hunk 3]
```diff
@@ -19,33 +19,44 @@ def f(x):
...
+    unique, counts = numpy.unique(site_class, return_counts=True)
+    print(dict(zip(unique, counts)), flush=True)
```
Comment (andrewkern) [inline benchmark/ts2fs.py:43]:
```suggestion
    # print(dict(zip(unique, counts)), flush=True)
```

File: benchmark/ts2fs.py [hunk 4]
```diff
@@ -19,33 +19,44 @@ def f(x):
...
-    mut_type = np.zeros(ts.num_sites)
+    # Mapping mutation type IDs to class of mutation (e.g., neutral, non-neutral)
+    mut_types = {}
+    for dfe in ts.metadata["stdpopsim"]["DFEs"]:
+        for mt in dfe["mutation_types"]:
+            mid = mt["slim_mutation_type_id"]
+            if not mid in mut_types:
+                mut_types[mid] = "neutral" if mt["is_neutral"] else "non_neutral"
+
+    print("These are the mutation types", flush=True)
```
Comment (andrewkern) [inline benchmark/ts2fs.py:30]:
```suggestion
    # print("These are the mutation types", flush=True)
```

File: benchmark/ts2fs.py [hunk 5]
```diff
@@ -19,33 +19,44 @@ def f(x):
...
+    print(mut_types, flush=True)
```
Comment (andrewkern) [inline benchmark/ts2fs.py:31]:
```suggestion
    # print(mut_types, flush=True)
```

File: benchmark/ts2fs.py (hunk 1 above)
Comment (mufernando) [reply benchmark/ts2fs.py:42]: oops!

File: benchmark/ts2fs.py [hunk 6]
```diff
@@ -19,33 +19,44 @@ def f(x):
...
-        else:
-            mut_type[j] = mt[0]
+        site_class[j] = mut_types[mt[0]] if len(mt) == 1 else "double_hit" 
+    assert sum(site_class == None) == 0
+    print("Number of sites per class is:", flush=True)
+    unique, counts = numpy.unique(site_class, return_counts=True)
+    print(dict(zip(unique, counts)), flush=True)
 
-    freqs = allele_counts(ts, [sample_sets])
+    freqs = allele_counts(ts, sample_sets)
```
Comment (andrewkern) [inline benchmark/ts2fs.py:45]:
```suggestion
    freqs = allele_counts(ts, [sample_sets])
```

**Pull request merged**